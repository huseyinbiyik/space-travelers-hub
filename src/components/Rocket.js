import React from 'react';

export default function Rocket() {
  return (
    <div>Rocket</div>
  );
}
