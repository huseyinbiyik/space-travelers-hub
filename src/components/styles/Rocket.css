.single-rocket {
  display: flex;
  justify-content: flex-start;
  gap: 20px;
}

.rocket-image {
  width: 300px;
  height: auto;
}

.rocket-title {
  margin-top: 0;
}

.rocket-desc {
  line-height: 25px;
}

.rocket-reserve-btn {
  height: 40px;
  color: white;
  background-color: #017bfe;
  border: none;
  padding: 8px 10px;
  border-radius: 3px;
  font-size: 15px;
}

.rocket-res-status {
  background-color: #1aa1b9;
  color: white;
  border-radius: 3px;
  padding: 2px;
  margin-right: 5px;
}

.rocket-cancel-btn {
  height: 40px;
  background-color: transparent;
  color: gray;
  border: 1px solid gray;
  padding: 8px 10px;
  border-radius: 3px;
  font-size: 15px;
}
