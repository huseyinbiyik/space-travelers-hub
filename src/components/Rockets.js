import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { fetchRockets } from '../redux/rockets/rockets';
import Rocket from './Rocket';
import './styles/Rockets.css';

export default function Rockets() {
  const rocketData = useSelector((state) => state.rockets);
  const dispatch = useDispatch();
  useEffect(() => {
    if (rocketData.length === 0) {
      dispatch(fetchRockets());
    }
  }, []);
  const rocketElements = rocketData.map((rocket) => (
    <Rocket
      key={rocket.id}
      id={rocket.id}
      img={rocket.img}
      description={rocket.description}
      title={rocket.rocket_name}
    />
  ));
  return (
    <div className="main-wrapper rocket-wrapper">
      {rocketElements}
    </div>
  );
}
