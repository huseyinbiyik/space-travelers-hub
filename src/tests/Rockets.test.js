import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import store from '../Redux/configureStore';
import Rockets from '../components/Rockets';
import Rocket from '../components/Rocket';
import rocketsReducer, { showRockets, addRocketReservation, cancelRocketReservation } from '../redux/rockets/rockets';
import rocketTestData from './testData';

describe('Rocket', () => {
  it('should render rocket component', () => {
    const rocket = render(
      <Provider store={store}>
        <Rocket />
      </Provider>,
    );
    expect(rocket).toMatchSnapshot();
  });

  it('should render rockets component', () => {
    const rockets = render(
      <Provider store={store}>
        <Rockets />
      </Provider>,
    );
    expect(rockets).toMatchSnapshot();
  });

  it('Should update the state with API data', () => {
    expect(rocketsReducer(rocketTestData, showRockets(rocketTestData)))

      .toEqual(rocketTestData);
  });

  it('Should update the state with correct id to reserved: true', () => {
    const rocketId = '1';
    expect(rocketsReducer(rocketTestData, addRocketReservation(rocketId)))
      .toEqual(rocketTestData.map((d) => (d.id === rocketId ? { ...d, reserved: true } : d)));
  });

  it('Should update the state with correct id to reserved: false', () => {
    const rocketId = '5';
    expect(rocketsReducer(rocketTestData, cancelRocketReservation(rocketId)))
      .toEqual(rocketTestData.map((d) => (d.id === rocketId ? { ...d, reserved: false } : d)));
  });
});
