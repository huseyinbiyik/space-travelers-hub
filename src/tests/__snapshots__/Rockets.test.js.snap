// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rocket should render rocket component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="single-rocket"
      >
        <div
          class="rocket-image-container"
        >
          <img
            class="rocket-image"
          />
        </div>
        <div
          class="rocket-desc-container"
        >
          <h2
            class="rocket-title"
          />
          <p
            class="rocket-desc"
          >
            <small
              class=""
            />
          </p>
          <button
            class="rocket-reserve-btn"
            type="submit"
          >
            Reserve Rocket
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="single-rocket"
    >
      <div
        class="rocket-image-container"
      >
        <img
          class="rocket-image"
        />
      </div>
      <div
        class="rocket-desc-container"
      >
        <h2
          class="rocket-title"
        />
        <p
          class="rocket-desc"
        >
          <small
            class=""
          />
        </p>
        <button
          class="rocket-reserve-btn"
          type="submit"
        >
          Reserve Rocket
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Rocket should render rockets component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="main-wrapper rocket-wrapper"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="main-wrapper rocket-wrapper"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
